klasses/api_test.py

```diff
@@ -174,6 +178,39 @@ def test_adjust_app_state_for_new_price(personal_price_amt, init_state, expected
     assert app.state == expected_state
 
 
+@factory.django.mute_signals(signals.post_save)
+@pytest.mark.django_db
+@pytest.mark.parametrize("order_status", [Order.CREATED, Order.FULFILLED])
+def test_adjust_app_state_for_new_price_with_existing_orders(order_status):
+    """
+    Test that adjust_app_state_for_new_price updates an application when personal price exists
+    and partial payment is made with an order in fulfilled status.
+    """
+    app = BootcampApplicationFactory.create(state=AppStates.AWAITING_PAYMENT.value)
+    run = app.bootcamp_run
+    user = app.user
+    personal_price = PersonalPriceFactory.create(price=990, user=user, bootcamp_run=run)
+    InstallmentFactory.create(bootcamp_run=run, amount=RUN_PRICE)
+    # Create payments such that the user has paid the original bootcamp run price
+    LineFactory.create_batch(
+        2,
+        order__user=user,
+        order__status=order_status,
+        bootcamp_run=run,
+        price=int(RUN_PRICE / 2),
```

Comment:
I think that this comment can be changed to something like 
`Create payments for the paid or pending order`

Reply:
Updated in https://github.com/mitodl/bootcamp-ecommerce/pull/1537/commits/3bb996c78a861b86aa9a7ebc2dd71c07f2714cdd